File: app/src/main/java/com/wizeline/daggerjava/di/AppComponent.java
package com.wizeline.daggerjava.di;

import android.app.Application;

import com.wizeline.daggerjava.ui.DaggerApp;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        AppBinder.class
})
public interface AppComponent {
    @Component.Builder
    public interface Builder {
        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }

    void inject(DaggerApp app);
}

File: app/src/main/java/com/wizeline/dagger/di/AppComponent.kt
package com.wizeline.dagger.di

import android.app.Application
import com.wizeline.dagger.view.DaggerApp
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule

@Component(
    modules = [
        AndroidInjectionModule::class,
        AppBinder::class,
        AppModule::class
    ]
)
interface AppComponent {
    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }

    fun inject(app: DaggerApp)
}